internal/finder/find.go

```diff
@@ -61,10 +63,96 @@ func addCloneToBlock(block *ast.BlockStmt, fields []*ast.Field) *ast.BlockStmt {
 	}
 }
 
+func migrateClone(typeName string, block *ast.BlockStmt, field *ast.Field, assigns []*ast.AssignStmt) *ast.BlockStmt {
+	list := make([]ast.Stmt, len(block.List))
+	copy(list, block.List)
+	var args []ast.Expr
+	var insert int
+
+	for i, stmt := range list {
+		as, ok := stmt.(*ast.AssignStmt)
+		if !ok {
+			continue
+		}
+		for _, assign := range assigns {
+			if as == assign {
+				sel, ok := assign.Lhs[0].(*ast.SelectorExpr)
+				if !ok {
+					continue
+				}
+				name := "new" + sel.Sel.Name
+				list[i] = &ast.AssignStmt{
+					Tok: token.DEFINE,
+					Lhs: []ast.Expr{
+						&ast.Ident{
+							Name: name,
+						},
+					},
+					Rhs: assign.Rhs,
+				}
+				args = append(args, &ast.KeyValueExpr{
+					Key: &ast.Ident{
+						Name: sel.Sel.Name,
+					},
+					Value: &ast.Ident{
+						Name: name,
+					},
+				})
+				insert = i
+			}
+		}
+	}
+	if len(args) > 0 {
+		insert++ // add one after
+		list = append(list[:insert], append([]ast.Stmt{&ast.AssignStmt{
+			Tok: token.DEFINE,
+			Lhs: []ast.Expr{
+				&ast.Ident{
+					Name: field.Names[0].Name,
+				},
+			},
+			Rhs: []ast.Expr{
+				&ast.CallExpr{
+					Fun: &ast.SelectorExpr{
+						X:   &ast.Ident{Name: "_" + field.Names[0].Name},
+						Sel: &ast.Ident{Name: "Apply"},
+					},
+					Args: []ast.Expr{
+						&ast.UnaryExpr{
+							X: &ast.CompositeLit{
+								Type: &ast.SelectorExpr{
+									X: &ast.Ident{
+										Name: "model",
+									},
+									Sel: &ast.Ident{
+										Name: typeName + "Patch",
+									},
+								},
+								Elts: args,
+							},
+							Op: token.AND,
+						},
+					},
+				},
+			},
+		},
+		}, list[insert:]...)...)
+	}
```

Comment:
I get what you are trying to do here @isacikgoz , but this would be MUCH simpler if you just used regular string concatenation here, without going through AST. It could be around 6-7 lines total :)

Reply:
Probably, I went too far with the `ast` 🥴, let's merge this one and we can replace it if this one gets to a limit. (probably it will)